apps/03TicTacToe.js

```diff
@@ -40,10 +44,38 @@ function printBoard() {
   console.log('2 ' + board[2].join(' | '));
 }
 
-function ticTacToe(row, column) {
+function getPrompt() {
+  console.log("===============================");
+  console.log("It's Player " + playerTurn + "'s turn.");
+  console.log();
+  printBoard();
+  prompt.get(['row', 'column'], function(error, result) {
+    var row = parseInt(result['row']);
+    var column = parseInt(result['column']);
+    //check for validity of row and column player move
+    if(validVal(row) && validVal(column) ){
+      ticTacToe(row, column);
+      getPrompt();
+    }
+    else{
+      console.log('invalid entry, try again');
+      getPrompt();
+    }
 
+    //prompt.get closes
+  });
+  //getPrompt() closes
+}
+
+function validVal(row_or_column){
+  return (validValues.indexOf(row_or_column) > -1 && validValues.indexOf(row_or_column) > -1)
```

Comment:
Having an array with the valid values is a good way to check!!! Its probably the most straight forward. Something else you could do:

```javascript
if(!isNaN(row_or_column) && row_or_column <=2 && row_or_column >=0)
```

I like your way better, I only leave this here to introduce isNaN, "is not an integer"

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN